### PR TITLE
Hasher benchmarks

### PR DESCRIPTION
**Pull request by drewstone**

### CODE REVIEW

File: standalone/runtime/Cargo.toml [hunk 1]
```diff
@@ -14,155 +14,169 @@ targets = ["x86_64-unknown-linux-gnu"]
...
 [dependencies]
-codec = { package = "parity-scale-codec", version = "2.3.0", default-features = false, features = ["derive", "max-encoded-len"] }
-hex-literal = { version = '0.3.1', optional = true }
-log = { version = "0.4.14", default-features = false }
-scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
-serde = { version = "1.0.101", optional = true, features = ["derive"] }
-static_assertions = "1.1.0"
+codec = {package = "parity-scale-codec", version = "2.3.0", default-features = false, features = ["derive", "max-encoded-len"]}
+# This getrandom package was having issues been compiled to wasm, that's why this override is here
+getrandom = {version = "0.2", features = ["js"]}
```
Comment (drewstone) [inline standalone/runtime/Cargo.toml:19]: This should never be in a runtime, `getrandom` doesn't compile to WASM that Substrate does afaik.

File: runtime/Cargo.toml [hunk 2]
```diff
@@ -20,6 +20,8 @@ log = { version = "0.4.14", default-features = false }
 scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
 serde = { version = "1.0.101", optional = true, features = ["derive"] }
 smallvec = "1.6.1"
+# This getrandom package was having issues been compiled to wasm in a downstream dependency, that's why this override is here
+getrandom = { version = "0.2", features = ["js"] }
```
Comment (drewstone) [inline runtime/Cargo.toml:24]: This should never be in a runtime, `getrandom` doesn't compile to WASM that Substrate does afaik.

File: standalone/runtime/Cargo.toml (hunk 1 above)
Comment (drewstone) [reply standalone/runtime/Cargo.toml:19]: The error you get often from the compiler is very misleading. When you do get WASM compilation errors it means that some other dependency is pulling in `std` features. The solution is not to add this from my experience.

File: runtime/Cargo.toml (hunk 2 above)
Comment (Wizdave97) [reply runtime/Cargo.toml:24]: I got an exact error that getrandom was not being compiled to wasm, getrandom can get compiled to wasm but the js feature needs to be enabled

**Comment by drewstone** [comment]
@Wizdave97 I fixed the build a least when omitting `standalone`. Will check tomorrow if that side still needs fixing but we should be able to run benchmarks now against the pallets.

As mentioned above, when you see an error like the one below, the solution is never to add `getrandom` this means that there is somewhere in the build where the `std` library is being imported when it should not. Turns out we were adding `frame-benchmarking/std` into the `pallet-hasher/runtime-benchmarks` which was causing this error. It is annoying that some rust errors don't directly communicate how to fix it, like this one, but do take note. Usually you can find these by slowly removing dependencies until you find the one that breaks things.
```
  219 | /         compile_error!("the wasm32-unknown-unknown target is not supported by \
  220 | |                         default, you may need to enable the \"js\" feature. \
  221 | |                         For more information see: \
  222 | |                         https://docs.rs/getrandom/#webassembly-support");
      | |_________________________________________________________________________^

  error[E0433]: failed to resolve: use of undeclared crate or module `imp`
     --> /Users/drewstone/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.3/src/lib.rs:246:5
      |
  246 |     imp::getrandom_inner(dest)
      |     ^^^ use of undeclared crate or module `imp`
```

**Comment by Wizdave97** [comment]
@drewstone weight functions have been added.

**Pull request merged**